File: leetcode/solutions/0035.SearchInsertPosition/searchInsert_test.go
package solution

import "testing"

func Test_0035_SearchInsertPosition(t *testing.T) {
	result := searchInsert([]int{1, 3, 5, 6}, 5)
	expected := 2

	if result != expected {
		t.Errorf("TEST FAILED! got: %#v, want: %#v.", result, expected)
	}

	result = searchInsert([]int{1, 3, 5, 6}, 2)
	expected = 1

	if result != expected {
		t.Errorf("TEST FAILED! got: %#v, want: %#v.", result, expected)
	}

	result = searchInsert([]int{1, 3, 5, 6}, 7)
	expected = 4

	if result != expected {
		t.Errorf("TEST FAILED! got: %#v, want: %#v.", result, expected)
	}

	result = searchInsert([]int{1, 3, 5, 6}, 0)
	expected = 0

	if result != expected {
		t.Errorf("TEST FAILED! got: %#v, want: %#v.", result, expected)
	}

}

File: leetcode/solutions/0035.SearchInsertPosition/searchInsert_test.ts
import { assertEquals } from "https://deno.land/std/testing/asserts.ts";
import searchInsert from "./searchInsert.ts";

Deno.test("0035. Search Insert Position", () => {
  assertEquals(searchInsert([1, 3, 5, 6], 5), 2);
  assertEquals(searchInsert([1, 3, 5, 6], 2), 1);
  assertEquals(searchInsert([1, 3, 5, 6], 7), 4);
  assertEquals(searchInsert([1, 3, 5, 6], 0), 0);
});
